_source/_posts/2023-07-31-workflows-workshop.md

```diff
@@ -0,0 +1,939 @@
+
+Each card represents one step in a flow. 
+
+A card has two types of fields: input and output. Input fields accept values. Output fields are values a card produces, which can be passed to subsequent cards.
+
+This is an example of the **Text - Length** card. This card calculates and returns the number of characters in a text string.
+
+- The **text** field is the card's input
+- The **length** field is the card's output (the lower grey area)
+
+{% img blog/workflows-workshop/Workflows_Text_Length_card.jpg alt:"The Text-Length card calculates text length" %}{: .center-image }
+
+There are two types of cards:
+
+- App action cards are steps that control other applications or web services. For example, Gmail, Slack, and Jira
+- Function cards are steps to interact with, change, or control the data in a flow. Some examples of functions cards:
+    - True/False: Evaluate values based on true or false conditions
+    -  Flow Control: Manage and manipulate the structure of your flow
+    -  Date & Time: Parse and manipulate times and dates
+    -  List: Create and iterate over lists of items
+    -  Number: Perform mathematical operations
+    -  Text: Build, modify, and parse text
+
+
+Here is again what you are going to build: 
+
+- Calling an API service to retrieve all the todo items
+- Creating a message from all the todo items
+- Emailing the message
+- Set up the flow to run on schedule
+
+Before building the flow, you need to get the Todo application.  
+
+## Setting up the Todo application
+In this section, you will complete the following steps:  
+
+1. Getting the Todo application
+2. Installing application dependencies
+3. Creating the database
+4. Adding an organziation to the application
+5. Running the application
+
+
+### Getting the Todo application
+In this step, you will get the Todo application.
+
+There are two ways to get the Todo application: 
+
+  _Option 1: Clone the repository_
+  1. Clone the [Todo application](https://github.com/oktadev/okta-enterprise-ready-workshops/) that already has OIDC support implemented. 
+  2. Run the following command `git checkout oidc-workshop-complete`
+  
+  _Option 2: Download the application_
+  1. Go to the [okta-enterprise-ready-workshops](https://github.com/oktadev/okta-enterprise-ready-workshops/tree/oidc-workshop-complete) GitHub page
+  2. Open the menu for the green **<> Code** button and select **Download Zip**
+
+### Installing application dependencies
+
+1. Navigate to the folder where you cloned or downloaded the Todo application
+    * If you downloaded the zip file, unzip the application file
+2. Run the following commands to install application dependencies:
+
+```
+npm ci
+npm install passport-http-bearer
+npm install @types/passport-http-bearer -D
+```
+
+### Creating the database
+
+Run the following command to generate a database and seed it with two users.  The user's name and password will write to the console.
+
+```
+npm run init-db
+```
+
+The users are 
```

Comment:
IMO I think instead of writing out the users and passwords here, we should tell people to look in the console for output. In case anything has to change.

Reply:
Updated.